src/Stratis.Bitcoin/Consensus/ConsensusManagerExtentions.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Consensus
+{
+    public static class ConsensusManagerExtentions
+    {
+        /// <summary>
+        /// Gets the chained block header given a block ID (hash).
+        /// </summary>
+        /// <param name="consensusManager">The consensus manager class.</param>
+        /// <param name="blockHash">Block hash to retrieve.</param>
+        /// <returns>The chained block header.</returns>
+        public static ChainedHeader GetBlock(this IConsensusManager consensusManager, uint256 blockHash)
```

Comment:
Here you are retrieving a whole block to get chained header from it. 
Also method should be async.